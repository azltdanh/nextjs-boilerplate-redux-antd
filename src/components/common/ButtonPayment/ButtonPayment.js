import { Row, Col, Button } from 'antd';

const ButtonPayment = ({ className, loading, total, sub, text, children }) => (
  <Row className={className}>
    <Col span={10}>
      <div className="sub-total">
        <b>{total}</b>
        <small>{sub}</small>
      </div>
    </Col>
    <Col span={14}>
      <Button
        size="large"
        type="default"
        htmlType="submit"
        block
        loading={loading}
        disabled={loading}
      >
        {text}
        {!text && children}
      </Button>
    </Col>
  </Row>
);

export default ButtonPayment;
