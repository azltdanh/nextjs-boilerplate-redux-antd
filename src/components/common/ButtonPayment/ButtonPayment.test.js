import ButtonPayment from './ButtonPayment';

describe(ButtonPayment.name, () => {
  it('renders', () => {
    const wrapper = shallow(<ButtonPayment>Confirm</ButtonPayment>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with props', () => {
    const wrapper = shallow(
      <ButtonPayment
        className="css-class"
        loading
        total="total"
        sub="sub"
        text="test"
      >
        Confirm
      </ButtonPayment>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
