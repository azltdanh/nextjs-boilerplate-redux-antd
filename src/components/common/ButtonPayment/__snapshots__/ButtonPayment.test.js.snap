// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPayment renders 1`] = `
<Row>
  <Col
    span={10}
  >
    <div
      className="sub-total"
    >
      <b />
      <small />
    </div>
  </Col>
  <Col
    span={14}
  >
    <Button
      block={true}
      ghost={false}
      htmlType="submit"
      loading={false}
      size="large"
      type="default"
    >
      Confirm
    </Button>
  </Col>
</Row>
`;

exports[`ButtonPayment renders with props 1`] = `
<Row
  className="css-class"
>
  <Col
    span={10}
  >
    <div
      className="sub-total"
    >
      <b>
        total
      </b>
      <small>
        sub
      </small>
    </div>
  </Col>
  <Col
    span={14}
  >
    <Button
      block={true}
      disabled={true}
      ghost={false}
      htmlType="submit"
      loading={true}
      size="large"
      type="default"
    >
      test
    </Button>
  </Col>
</Row>
`;
